libnavigation-metrics/src/main/java/com/mapbox/navigation/metrics/MapboxMetricsReporter.kt

```diff
@@ -2,29 +2,50 @@ package com.mapbox.navigation.metrics
 
 import android.content.Context
 import com.google.gson.Gson
-import com.mapbox.android.telemetry.MapboxTelemetry
+import com.mapbox.bindgen.Value
+import com.mapbox.common.Event
+import com.mapbox.common.EventPriority
+import com.mapbox.common.EventsServerOptions
+import com.mapbox.common.EventsServiceError
+import com.mapbox.common.EventsServiceInterface
+import com.mapbox.common.EventsServiceObserver
 import com.mapbox.navigation.base.metrics.MetricEvent
 import com.mapbox.navigation.base.metrics.MetricsObserver
 import com.mapbox.navigation.base.metrics.MetricsReporter
-import com.mapbox.navigation.metrics.extensions.toTelemetryEvent
+import com.mapbox.navigation.metrics.internal.EventsServiceProvider
+import com.mapbox.navigation.metrics.internal.TelemetryUtilsDelegate
 import com.mapbox.navigation.utils.internal.InternalJobControlFactory
+import com.mapbox.navigation.utils.internal.logD
+import com.mapbox.navigation.utils.internal.logE
 import kotlinx.coroutines.cancelChildren
 import kotlinx.coroutines.launch
 
 /**
  * Default implementation of [MetricsReporter] interface.
  */
 object MapboxMetricsReporter : MetricsReporter {
+    private const val LOG_CATEGORY = "MapboxMetricsReporter"
 
     private val gson = Gson()
-    private lateinit var mapboxTelemetry: MapboxTelemetry
+    private lateinit var eventsService: EventsServiceInterface
```

Comment:
We also need `TelemetryService` here which should be just initialized with the same `EventsServerOptions`

We use `EventsService` to send interaction events (like `navigation_arrive`, `navigation_*`, `turnstile`, `map_load`, etc.) which we are sending manually.

`TelemetryService` sends `location` and `metrics` which do not need to be triggered manually (require just initialized and alive `TelemetryService`)